app/handlers/profiles/settings/username.go

```diff
@@ -25,56 +25,67 @@ func NameChangeSubmitHandler(c *gin.Context) {
 		return
 	}
 
-	if ctx.User.Privileges&common.UserPrivilegeDonor == 0 {
-		tu.Resp403(c)
+	var hasFreeUsernameChange bool
+	sql_err := services.DB.QueryRow("SELECT 1 FROM users WHERE id = ? AND has_free_username_change = 1", ctx.User.ID).Scan(&hasFreeUsernameChange)
+	if sql_err != nil && sql_err != sql.ErrNoRows {
+		slog.Error("Error querying has_free_username_change", "user", ctx.User.ID)
+		sessions.AddMessage(c, msg.ErrorMessage{lu.T(c, "Something went wrong.")})
+		sessions.GetSession(c).Save()
+		c.Redirect(302, "/u/"+strconv.Itoa(int(ctx.User.ID)))
 		return
 	}
 
-	if c.PostForm("name") != "" {
-		username := strings.TrimSpace(c.PostForm("name"))
-		// check if username already taken
-		if services.DB.QueryRow("SELECT 1 FROM users WHERE username_safe = ?", uu.SafeUsername(username)).
-			Scan(new(int)) != sql.ErrNoRows {
-			sessions.AddMessage(c, msg.ErrorMessage{lu.T(c, "Username taken.")})
-			sessions.GetSession(c).Save()
-
-			c.Redirect(302, "/u/"+strconv.Itoa(int(sessions.GetContext(c).User.ID)))
+	isDonor := (ctx.User.Privileges & common.UserPrivilegeDonor) != 0
+	if !isDonor {
+		if !hasFreeUsernameChange {
+			tu.Resp403(c)
 			return
```

Comment:
it seems we're showing this tab to users who might not have a username change available, but will this case present them with a clear error message? "You have already used your free name change; donator required" or whatever

Reply:
either that or make it clear as a req to show the tab; something to show users they're only able to get one for free